### PR TITLE
Escaping block character for file match

### PR DESCRIPTION
**Pull request by vinodkumar3**
Issue - #11673 
Here is documentation [reference](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/file-matching-patterns?view=azure-devops#escaping).

### CODE REVIEW

File: Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts [hunk 1]
```diff
@@ -278,7 +278,7 @@ export class ResourceGroup {
     private createDeploymentName(): string {
         var name: string;
         if (this.taskParameters.templateLocation == "Linked artifact") {
-            name = tl.findMatch(tl.getVariable("System.DefaultWorkingDirectory"), this.taskParameters.csmFile)[0];
+            name = tl.findMatch(tl.getVariable("System.DefaultWorkingDirectory"), this.taskParameters.csmFile.replace(/[\[]/g, '$&[]'))[0];
```
Comment (jalajmsft) [inline Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts]: Should we have a common function for this logic?

Comment (jalajmsft) [inline Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts]: Is bracket only the speacial character allowed in filename?

Comment (bishal-pdMSFT) [inline Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts]: the replacement can be done while populating `this.taskParameters.csmFile` at the beginning of the task. This way you do not need to replace at multiple places.

File: Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts [hunk 2]
```diff
@@ -405,7 +405,7 @@ export class ResourceGroup {
 
         var parameters: Map<string, ParameterValue> = {} as Map<string, ParameterValue>;
         if (utils.isNonEmpty(this.taskParameters.csmParametersFile)) {
-            var fileMatches = tl.findMatch(tl.getVariable("System.DefaultWorkingDirectory"), this.taskParameters.csmParametersFile);
+            var fileMatches = tl.findMatch(tl.getVariable("System.DefaultWorkingDirectory"), this.taskParameters.csmParametersFile.replace(/[\[]/g, '$&[]'));
```
Comment (bishal-pdMSFT) [inline Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts]: same for parameter file name

File: Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts (hunk 1 above)
Comment (vinodkumar3) [reply Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts]: We might do non globbing operations as well on these file paths so its not a good idea to modify them at starting of the tasks. Thoughts ?

Comment (vinodkumar3) [reply Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts]: Yes, others are * and ? which are not allowed in file paths

Comment (bishal-pdMSFT) [reply Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts]: My intention is that we do not repeat the same logic multiple times. Another option is to add a common function call from all these places.

Comment (vinodkumar3) [reply Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts]: Done

**Pull request merged**